lib/attribute.js

```diff
@@ -0,0 +1,407 @@
+   */
+  set: function(name, val, opts) {
+    return this._setAttr(name, val, opts);
+  },
+
+  /**
+   * Like `set`, but allows setting of readOnly/writeOnce attributes.
+   *
+   * @method _set
+   * @protected
+   * @chainable
+   *
+   * @param {String} name The name of the attribute.
+   * @param {Any} val The value to set the attribute to.
+   * @param {Object} [opts] Optional data providing the circumstances for the change.
+   */
+  _set: function(name, val, opts) {
+    return this._setAttr(name, val, opts, true);
+  },
+
+  /**
+   * Set a whole buncha attrs.
+   *
+   * Fires the generic change event's "on" before all and "after" after all are set.
+   * This is the least noisy way to set several attributes from the point of view of listeners of "change."
+   * @method addAttrs
+   * @param  {Object} name:value
+   */
+  setAttrs: function(attrs) {
+    var newVals = {};
+    var prevVals = {};
+    ObjectUtils.each(attrs, function(value, name) {
+      var newVal = this._prepareValue(name, value);
+      if (newVal !== FAIL) {
+        newVals[name] = value;
+        prevVals[name] = this.get(name);
+      }
+    }, this);
+
+    var adtlEventParams = {
+      newVals: newVals,
+      prevVals: prevVals,
+      multi: true
+    };
+
+    this.fire('change', function() {
+      ObjectUtils.each(newVals, function(newVal, name) {
+
+        var adtlSpecificEventParams = {
+          prevVal: prevVals[name],
+          newVal: newVal,
+          name: name
+        };
+        
+        this.fire(name + 'Change', function() {
+          this._setValueHelper(name, newVal);
+        }.bind(this), adtlSpecificEventParams);
+
+      }, this);
+    }.bind(this), adtlEventParams);
+
+    return this;
+  },
+
+  /**
+   * Common method to set the value of an attribute
+   * @method  _set
+   * @param   {String} name
+   * @param   value
+   * @param   [opts] can be anything. Passed to validator, setter.
+   * @param   {Boolean} [force] bypass readOnly and writeOnce checks
+   * @param   {Boolean} [onInit] from the constructor. don't count againt writeOnce, no events
+   * @private
+   */
+  _setAttr: function(name, value, opts, force, onInit) {
+    value = this._prepareValue(name, value, opts, force, onInit);
+    if (value === FAIL) {
+      return;
+    }
+
+    var adtlEventParams = {
+      prevVal: this.get(name),
+      newVal: value,
+      name: name
+    };
+
+    /**
+    * Two events are fired: `change` and `{attributeName}Change`
+    * @event change
+    * @property {any} prevVal the old value
+    * @property {any} nevVal the nev value
+    * @property {String} name the attr name
+    */
+    this.fire('change', function() {
+      this.fire(name + 'Change', function() {
+        this._setValueHelper(name, value, onInit);
+      }.bind(this), adtlEventParams);
+    }.bind(this), adtlEventParams);
+
+    return this;
+  },
+
+  _prepareValue: function(name, value, opts, force, onInit) {
```

Comment:
Well, local function variables would make these methods *actually* invisible to the inheriting class. I like the idea.